# Notebook pour interagir avec l'API Gemini de Google

------------------------------------------------
ÉTAPE 1: Installation des bibliothèques nécessaires
------------------------------------------------
Installer le SDK Python de Google AI avec pip

In [1]:
!pip install -q google-generativeai

------------------------------------------------
# ÉTAPE 2: Importation des modules nécessaires
------------------------------------------------

In [9]:
import os
import google.generativeai as genai
from IPython.display import Markdown, display
from google.colab import userdata


------------------------------------------------
# ÉTAPE 3: Configuration de l'authentification API
------------------------------------------------
Récupérer la clé API de manière sécurisée depuis les secrets de Colab
Vous devez d'abord enregistrer votre clé dans l'outil de secrets de Colab

In [10]:
api_key = userdata.get('GEMINI_API_KEY')

print('API KEY is OK')

# Alternative moins sécurisée : définir directement la clé API (déconseillé)
# api_key = "VOTRE_CLÉ_API"  # Remplacez par votre clé API réelle

API KEY is OK


------------------------------------------------
# ÉTAPE 4: Configuration de la bibliothèque d'IA générative
------------------------------------------------
Initialiser la bibliothèque avec votre clé API

In [11]:
genai.configure(api_key=api_key)

------------------------------------------------
# ÉTAPE 5: Définition de la fonction pour interroger Gemini
------------------------------------------------

In [12]:
def get_gemini_response(prompt_text, model="gemini-1.5-pro"):
    """
    Obtenir une réponse du modèle Gemini.

    Args:
        prompt_text (str): Le texte de la requête à envoyer
        model (str): Le modèle Gemini à utiliser

    Returns:
        La réponse générée par l'API
    """
    # Initialiser le modèle
    model = genai.GenerativeModel(model)

    # Générer le contenu
    response = model.generate_content(prompt_text)

    return response

------------------------------------------------
# ÉTAPE 6: Fonction principale d'exécution
------------------------------------------------

In [17]:
def main():


    # La requête est maintenant stockée directement dans une variable
    # prompt_example = """What are the 5 facts to know about Costa-Rica?"""
    # Autre exemple de requête (commenté)
    # prompt_example = """Explain quantum computing in simple terms."""

    prompt_example = """


    En tant que journaliste propose-moi 5 titres alternatifs en prenant exemple comme modèle
   'Chinois capturés par l'Ukraine : Pékin répond à Zelensky et dénonce des accusations "sans fondement"'
   avec les meilleures règles SEO possibles pour le référencement sur Discover de Google
   Indique aussi les bonnes pratiques SEO pour performer sur Discover de Google


    """

    # Cette ligne n'est plus nécessaire car nous utilisons la variable directement
    # prompt_filename = "prompt_example.txt"

    # Charger la requête (maintenant directement depuis la variable)
    prompt_text = prompt_example
    # Cette ligne est remplacée
    # prompt_text = load_prompt(prompt_filename)

    print(f"Requête chargée: {prompt_text}\n")
    print("Envoi à l'API Gemini...\n")

    # Obtenir la réponse de Gemini
    try:
        # Appel à l'API Gemini
        response = get_gemini_response(prompt_text)

        # Afficher la réponse
        print("Réponse de Gemini:")
        print("-" * 50)
        display(Markdown(response.text))
        print("-" * 50)
    except Exception as e:
        # Gestion des erreurs
        print(f"Erreur: {e}")
        print("Si vous voyez une erreur de clé API, assurez-vous de configurer correctement votre clé API.")
        print("Instructions: Allez sur Google AI Studio pour obtenir une clé API")
        print("Ensuite dans Colab, cliquez sur l'icône 🔑 dans la barre latérale gauche pour l'enregistrer comme GEMINI_API_KEY")

------------------------------------------------
# ÉTAPE 7: Point d'entrée du script
------------------------------------------------
Exécuter la fonction principale si ce script est exécuté directement


In [18]:
if __name__ == "__main__":
    main()

Requête chargée: 
    
    
    En tant que journaliste propose-moi 5 titres alternatifs en prenant exemple comme modèle
   'Chinois capturés par l'Ukraine : Pékin répond à Zelensky et dénonce des accusations "sans fondement"'
   avec les meilleures règles SEO possibles pour le référencement sur Discover de Google
   Indique aussi les bonnes pratiques SEO pour performer sur Discover de Google


    

Envoi à l'API Gemini...

Réponse de Gemini:
--------------------------------------------------


Voici 5 titres alternatifs optimisés pour Google Discover, inspirés de l'exemple :

1. **Ukraine accuse la Chine de capture : Réponse cinglante de Pékin** (Court, percutant, mots-clés importants en début de phrase)
2. **Prisonniers chinois en Ukraine ? La Chine dément les accusations de Zelensky** (Question intrigue, mots-clés importants, mentionne les acteurs principaux)
3. **Conflit Ukraine : Pékin nie toute capture de citoyens chinois, Zelensky sous le feu des critiques**  (Contexte plus large, mots-clés pertinents, élément de suspense)
4. **Tension Chine-Ukraine : La vérité sur les prétendus prisonniers chinois** (Clair, concis, évoque l'actualité et le mystère)
5. **Zelensky vs Pékin : Accusations de capture de Chinois, la Chine riposte** (Opposition directe, noms importants, sujet controversé)


**Bonnes pratiques SEO pour performer sur Google Discover :**

* **Titres accrocheurs et concis:**  Privilégiez des titres courts (moins de 70 caractères), percutants et suscitant la curiosité.  Utilisez des verbes d'action et posez des questions pour inciter au clic.
* **Mots-clés pertinents:**  Intégrez les mots-clés les plus importants au début du titre pour une meilleure visibilité. Pensez aux termes que les utilisateurs rechercheraient sur Google.  Dans cet exemple : "Chine," "Ukraine," "Zelensky," "prisonniers," "capture."
* **Contenu de haute qualité:** Google Discover privilégie le contenu original, pertinent, informatif et bien écrit. Assurez-vous que l'article est complet, précis et étayé par des sources fiables.
* **Images et vidéos de haute qualité:** Les visuels attrayants sont essentiels pour attirer l'attention sur Google Discover. Utilisez des images et des vidéos de grande taille, en haute résolution et optimisées pour le web.  Favorisez un format 1200 x 628 pixels.
* **Optimisation E-A-T:**  Google accorde de l'importance à l'expertise, l'autorité et la fiabilité (E-A-T).  Assurez-vous que votre site web est crédible, que l'auteur de l'article est reconnu dans son domaine et que les informations sont vérifiées.
* **Contenu evergreen (si possible):**  Bien que l'actualité soit importante pour Discover, le contenu "evergreen" (intemporel) peut continuer à générer du trafic sur le long terme.
* **Fréquence de publication régulière:** Publier régulièrement du contenu frais et de qualité signale à Google que votre site est actif et pertinent.
* **Données structurées:**  Utilisez des données structurées pour aider Google à comprendre le contenu de votre page et à l'afficher correctement dans Discover.
* **AMP (Accelerated Mobile Pages):**  Les pages AMP se chargent plus rapidement sur les appareils mobiles, ce qui est un facteur important pour Google Discover.
* **Analyser les performances:**  Utilisez Google Search Console et Google Analytics pour suivre les performances de votre contenu sur Discover et identifier les axes d'amélioration.  Regardez les données de clics, impressions et temps de lecture.

En appliquant ces bonnes pratiques, vous maximiserez vos chances d'apparaître et de performer sur Google Discover.


--------------------------------------------------


------------------------------------------------
# ÉTAPE 8: Base de connaissances
------------------------------------------------
**Bonnes pratiques SEO pour performer sur Google Discover :**




- **Titres accrocheurs et concis:** Privilégiez des titres courts (moins de 70 caractères), percutants et suscitant la curiosité. Utilisez des verbes d'action et posez des questions pour inciter au clic.

- **Mots-clés pertinents:** Intégrez les mots-clés les plus importants au début du titre pour une meilleure visibilité. Pensez aux termes que les utilisateurs rechercheraient sur Google. Dans cet exemple : "Chine," "Ukraine," "Zelensky," "prisonniers," "capture."

- **Contenu de haute qualité:** Google Discover privilégie le contenu original, pertinent, informatif et bien écrit. Assurez-vous que l'article est complet, précis et étayé par des sources fiables.

- **Images et vidéos de haute qualité:** Les visuels attrayants sont essentiels pour attirer l'attention sur Google Discover. Utilisez des images et des vidéos de grande taille, en haute résolution et optimisées pour le web. Favorisez un format 1200 x 628 pixels.

- **Optimisation E-A-T:** Google accorde de l'importance à l'expertise, l'autorité et la fiabilité (E-A-T). Assurez-vous que votre site web est crédible, que l'auteur de l'article est reconnu dans son domaine et que les informations sont vérifiées.
Contenu evergreen (si possible): Bien que l'actualité soit importante pour Discover, le contenu "evergreen" (intemporel) peut continuer à générer du trafic sur le long terme.

- **Fréquence de publication régulière:** Publier régulièrement du contenu frais et de qualité signale à Google que votre site est actif et pertinent.

- **Données structurées:** Utilisez des données structurées pour aider Google à comprendre le contenu de votre page et à l'afficher correctement dans Discover.

- **AMP (Accelerated Mobile Pages):** Les pages AMP se chargent plus rapidement sur les appareils mobiles, ce qui est un facteur important pour Google Discover.

- **Analyser les performances:** Utilisez Google Search Console et Google Analytics pour suivre les performances de votre contenu sur Discover et identifier les axes d'amélioration. Regardez les données de clics, impressions et temps de lecture.



